### PR TITLE
One jest to rule them all

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Adds a `yarn tdd` (Test Driven Development) to start jest in watch mode
- Consolidates jest config in a single file on the monorepo root
- Adds jest plugins to enhance watch mode

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] yarn tdd
- [ ] yarn test

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      modulePathIgnorePatterns: [
+        //
+        "<rootDir>/apps/web/test/__fixtures__",
+        "<rootDir>/apps/web/node_modules",
+        "<rootDir>/apps/web/dist",
+      ],
+      clearMocks: true,
+      setupFilesAfterEnv: ["<rootDir>/tests/config/singleton.ts"],
+      setupFiles: ["<rootDir>/apps/web/test/jest-setup.js"],
+      testMatch: ["**/test/lib/**/*.(spec|test).(ts|tsx|js)", "**/__tests__/**/*.(spec|test).(ts|tsx|js)"],
```
Comment (alannnc) [inline jest.config.ts:20]: I would go only leaving option two, and renaming actual test/lib to _tests_.
```suggestion
      testMatch: ["**/__tests__/**/*.(spec|test).(ts|tsx|js)"],
```

**Review by agustif — APPROVED** [review]
LGTM! 
This is much faster to run than E2E, I like jest!

image

**Comment by zomars** [comment]
> This is much faster to run than E2E, I like jest!

This is not a replacement for E2E tho, but important nonetheless

File: jest.config.ts (hunk 1 above)
Comment (zomars) [reply jest.config.ts:20]: Good for a followup

**Pull request merged**